packages/react-ui-core/src/Counter/Counter.js

```diff
@@ -20,6 +20,7 @@ export default class Counter extends PureComponent {
     min: PropTypes.number,
     max: PropTypes.number,
     className: PropTypes.string,
+    textTid: PropTypes.string,
```

Comment:
lets call this tid as a generic prop for test id

Reply:
The reason I called it textTid is that it specifically refers to the text that will display for this instance of the counter component - not a generic tid that the whole component would use.

I guess I could give it a generic tid name, though, and then derive the text data-tid from that

Reply:
We shouldn't be adding any specific nightwatch related data to these components. That's not what react-ui-core is about.  The better thing to do here would be to _not_ pass in a `data-tid`, but rather spread the props on the outer most div.  That way the component (without having to explicitly pass it) now has the `data-tid`.  If you _really_ need something on a child node, then we can add something more static like `data-counter="text"` or something. I can talk this over when you're ready.

Reply:
oh i didn't realize that - ideally the TID sits on the outside and you use that to climb into the HTML so we are not adding attributes needlessly - is this possible? 

eg `document.querySelector(someTid).text()`